Tests/NumLib/NaturalNodeCoordinates.cpp

```diff
@@ -0,0 +1,363 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <vector>
+
+#include "MeshLib/ElementCoordinatesMappingLocal.h"
+
+#include "NumLib/Fem/CoordinatesMapping/NaturalNodeCoordinates.h"
+#include "NumLib/Fem/CoordinatesMapping/ShapeMatrices.h"
+#include "NumLib/Fem/FiniteElement/C0IsoparametricElements.h"
+#include "NumLib/Fem/Integration/GaussIntegrationPolicy.h"
+#include "NumLib/Fem/ShapeMatrixPolicy.h"
+
+template <typename ShapeFunction, int GlobalDim>
+bool test(MeshLib::Element const& element)
+{
+    using ShapeMatricesType = ShapeMatrixPolicyType<ShapeFunction, GlobalDim>;
+    using FemType =
+        NumLib::TemplateIsoparametric<ShapeFunction, ShapeMatricesType>;
+
+    // Test if the cast is possible.
+    bool const dynamic_cast_failed =
+        dynamic_cast<typename ShapeFunction::MeshElement const*>(&element) ==
+        nullptr;
+    if (dynamic_cast_failed)
+    {
+        EXPECT_TRUE(dynamic_cast_failed);
```

Comment:
Minor: Maybe just remove that?

Reply:
While the implementation of the tests I run into a mismatch somehow. Would like to keep it...
The EXPECT_TRUE gives a good message on what failed, whereas return false does not give this hint.

Reply:
But in this branch `dynamic_cast_failed` will always be true, as checked by `if (dynamic_cast_failed)`. Or am I missing something?

Reply:
Yes, it will be true and a message will be shown while the test is running. W/o this expect only error message in the main test will be shown. It's just for message.

Reply:
I see now :man_facepalming: . My bad! I'll fix it.